File: python-examples/abstraction/BookShelf.py
from Book import Book

def main():
    book1 = Book("Intro to Dynamic Programming", "Tim Budd", 1991)
    book1.print_details()

    book1.title = "Intro to OOP"
    book1.print_details()

    print(f"Total books: {Book.get_book_count()}")

if __name__ == "__main__":
    main()

File: cpp-examples/abstraction/BookShelf.cpp
#include "Book.h"
#include <iostream>

int main() {
    Book myBook ("An Introduction to Dynamic Programming", "Timothy Budd", 1991);
    myBook.printDetails();

    myBook.setTitle("An Introduction to Object-Oriented Programming");
    myBook.printDetails();

    std::cout << "Total books: " << Book::getBookCount() << std::endl;

    return 0;
}